cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -105,7 +105,19 @@ public void getSystemProfile(HttpRequest request, HttpResponder responder,
   public void writeSystemProfile(FullHttpRequest request, HttpResponder responder,
                                  @PathParam("profile-name") String profileName) throws Exception {
     ProfileId profileId = getValidatedProfile(NamespaceId.SYSTEM, profileName);
-    accessEnforcer.enforce(profileId, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+    boolean profileExists;
+    try {
+      profileService.getProfile(profileId);
+      profileExists = true;
+    } catch (NotFoundException e) {
+      profileExists = false;
+    }
+    if (profileExists) {
+      accessEnforcer.enforce(profileId, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+    } else {
+      accessEnforcer.enforceOnParent(EntityType.PROFILE, profileId.getParent(), authenticationContext.getPrincipal(),
```

Comment:
As soon as you know the profile name you can check it with `enforce`, not `enforceOnParent`

Reply:
Fixed